client/lib/viewport/react-helpers.js

```diff
@@ -0,0 +1,121 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { useState, useEffect } from 'react';
+import { camelCase, upperFirst } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	isWithinBreakpoint,
+	addWithinBreakpointListener,
+	MOBILE_BREAKPOINT,
+	DESKTOP_BREAKPOINT,
+} from './index';
+
+/**
+ * React hook for getting the status for a breakpoint and keeping it updated.
+ *
+ * @param {String} breakpoint The breakpoint to consider.
+ *
+ * @returns {Boolean} The current status for the breakpoint.
+ */
+export function useBreakpoint( breakpoint ) {
+	const [ isActive, setIsActive ] = useState( () => isWithinBreakpoint( breakpoint ) );
+
+	function handleBreakpointChange( currentStatus ) {
+		setIsActive( currentStatus );
+	}
+
+	useEffect(() => {
+		const unsubscribe = addWithinBreakpointListener( breakpoint, handleBreakpointChange );
+
+		return function cleanup() {
+			if ( unsubscribe ) {
+				unsubscribe();
+			}
+		};
```

Comment:
This could be much simpler if `unsubscribe` was always a callable function. `noop` in failed cases.

Reply:
Yes, and we could even return `unsubscribe` itself, without needing to wrap it in an arrow function 👍